dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/AbstractClient.java

```diff
@@ -931,29 +931,25 @@ protected void setConfiguration(ClientConfiguration config) {
     // invocation thus it is also important to have baseURI and currentURI
     // synched up with the latest endpoint address, after a successful proxy
     // or web client invocation has returned
-    protected void prepareConduitSelector(Message m, URI currentURI, boolean proxy) {
+    protected void prepareConduitSelector(Message message, URI currentURI, boolean proxy) {
         try {
-            cfg.prepareConduitSelector(m);
+            cfg.prepareConduitSelector(message);
 
         } catch (Fault ex) {
             LOG.warning("Failure to prepare a message from conduit selector");
         }
-        //Liberty code change start
-        MessageImpl message = (MessageImpl) m;
         message.getExchange().put(ConduitSelector.class, cfg.getConduitSelector());
         message.getExchange().put(Service.class, cfg.getConduitSelector().getEndpoint().getService());
 
-        String address = (String)message.getEndpointAddress();
+        String address = (String)message.get(Message.ENDPOINT_ADDRESS);
         // custom conduits may override the initial/current address
         if (address.startsWith(HTTP_SCHEME) && !address.equals(currentURI.toString())) {
             URI baseAddress = URI.create(address);
             currentURI = calculateNewRequestURI(baseAddress, currentURI, proxy);
-            String uri = currentURI.toString();
-            message.setEndpointAddress(uri);
-            message.setRequestUri(uri);
+            message.put(Message.ENDPOINT_ADDRESS, currentURI.toString());
+            message.put(Message.REQUEST_URI, currentURI.toString());
```

Comment:
for this change, I'd probably leave the `String uri...` line, so something like:
```
String uri = currentURI.toString();
message.put(Message.ENDPOINT_ADDRESS, uri);
message.put(Message.REQUEST_URI, uri);
```

that way, we're only calling `uri.toString()` once.